packages/platform-sdk-ada/src/services/transaction.ts

```diff
@@ -44,26 +47,41 @@ export class TransactionService implements Contracts.TransactionService {
 		);
 
 		// Get a `Bip32PrivateKey` instance according to `CIP1852` and turn it into a `PrivateKey` instance
-		const privateKey = getCip1852Account(
-			input.sign.mnemonic,
-			this.#config.get<number>("network.crypto.slip44"),
-		).to_raw_key();
+		const accountKey: Bip32PrivateKey = deriveAccountKey(deriveRootKey(input.sign.mnemonic), 0);
+		const publicKey = accountKey.to_public();
 
-		// These are the inputs (UTXO) that will be consumed to satisfy the outputs. Any change will be transfered back to the sender
-		const utxos: any = await this.listUnspentTransactions(input.from);
+		/* TODO: We need to determine how to specify the input.
+			 input.from is currently used as an address, but in Cardano it should be more like
+			 transferring from a particular wallet to a destination address (or may be addresses).
+			 So we need to look at all spendable utxos from the wallet, not just from a source address,
+			 because the wallet may have enough founds but spread through different addresses,
+			 both internal (change) and external (spend).
+		 */
+		const utxos: UnspentTransaction[] = await this.listUnspentTransactions(input.from); // when more that one utxo, they seem to be ordered by amount descending
 
+		// Figure out the utxos to use
+		// TODO Need to make sure it covers the fess also. We need to be clever here.
+		const usedUtxos: UnspentTransaction[] = [];
+		const requestedAmount: number = parseInt(input.data.amount); // TODO see if we need to use bigint here
+		let amount: number = 0;
 		for (let i = 0; i < utxos.length; i++) {
```

Comment:
@faustbrian We need to do better than this!!! Let's suppose we have several utxos, we need to come up with some way to pick the best set that covers the amount to transfer + fees.
I saw some wallets offer to use oldest received first, minimize fees, etc. Not sure how we want to approach the utxo selection. How is it for Bitcoin.

Reply:
I'll check yoroi and see how they select the UTXO that are consumed.